tasks/Task.cpp

```diff
@@ -216,25 +216,30 @@ void Task::updateHook()
 				  size = _marker_size.get();
 				}
 
-				//estimate pose and push_back to rbs_vector
-				base::samples::RigidBodyState rbs;
-				getRbs(rbs, size, det->p, camera_k, cv::Mat());
-				rbs.sourceFrame = getMarkerFrameName( det->id);
-                                rbs.targetFrame = _camera_frame_name.value();
-				rbs.time = current_frame_ptr->time;
-				rbs_vector.push_back(rbs);
-
-				std::cout << "APRILTAGS ID: " << det->id <<
-						" x: " << rbs.position.x() <<
-						" y: " << rbs.position.y() <<
-						" z: " << rbs.position.z() <<
-						" roll: "  << rbs.getRoll()*180/M_PI  <<
-						" pitch: " << rbs.getPitch()*180/M_PI <<
-						" yaw: "   << rbs.getYaw()*180/M_PI   <<
-						" undistort_time: " << t_undistort*1000 << " ms " <<
-						" prepare_time: " << t_prepare*1000 << " ms" <<
-						" extract_time: " << t_detect*1000 << " ms" <<
-						std::endl;
+				//Compute pose
+                if(_pose_calculation.get())
+                {
+                    base::samples::RigidBodyState rbs;
+                    getRbs(rbs, size, det->p, camera_k, cv::Mat());
+
+                    //set the camera frame as the reference frame
+                    rbs.sourceFrame = getMarkerFrameName(det->id);
+                    rbs.targetFrame = _camera_frame.value();
+                    rbs.time = current_frame_ptr->time;
+                    rbs_vector.push_back(rbs);
+
+                    std::cout << "APRILTAGS ID: " << det->id <<
+                        " x: " << rbs.position.x() <<
+                        " y: " << rbs.position.y() <<
+                        " z: " << rbs.position.z() <<
+                        " roll: "  << rbs.getRoll()*180/M_PI  <<
+                        " pitch: " << rbs.getPitch()*180/M_PI <<
+                        " yaw: "   << rbs.getYaw()*180/M_PI   <<
+                        " undistort_time: " << t_undistort*1000 << " ms " <<
+                        " prepare_time: " << t_prepare*1000 << " ms" <<
+                        " extract_time: " << t_detect*1000 << " ms" <<
+                        std::endl;
```

Comment:
For the future try to make use of logging libraries, which allows the user to set the log level. For instance this would be an output for a debug level. Candidates are base-lib (http://rock-robotics.org/master/documentation/debugging/index.html), glog or the RTT logging.